Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2215,7 +2234,40 @@ else if (!(req instanceof ServerRequestInitSession) && (!hasSession() || !hasDev
             e.printStackTrace();
         }
     }
-    
+
+    private void processNextQueueItemForced() {
+
+        intentState_ = INTENT_STATE.READY;
+
+        if (BranchUtil.isTestModeEnabled(context_)) {
+            prefHelper_.setExternDebug();
+        }
+
+        startSessionForced();
```

Comment:
This is pretty weird a for a few reasons:
1. You're calling this every time in the implementation of process queue forced? Not sure why you would do that more generically in this function.
2. I don't understand why you need to trigger `initUserSessionInternal` 2X. You've already hit this function in `initSessionForced`...

Reply:
Thanks for these points! Have refreshed the implementation.